### ISSUE TITLE
Bug: TypeError for URL property with empty link

### ISSUE THREAD

**Issue by kamre** [mattermost/focalboard#3751]
## Steps to reproduce the behavior

1. Create new empty board
2. Create new card
3. Add URL property to card
4. Close card dialog
5. Show URL property on the board view
6. Board view is not visible and there is an error in the console:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'trim')
    at URLProperty (url.tsx:93:76)
```

## Expected behavior

Board view with the card should be visible.

## Edition and Platform

 - Edition: Personal Server
 - Version: `main` branch
 - Browser and OS: Chrome on Mac

## Additional context (optional)

Sev 1: Affects critical functionality without a workaround

Looks like a regression after recent properties code refactoring

**Comment by kamre** [comment]
Possible fix:
```diff
diff --git a/webapp/src/properties/url/url.tsx b/webapp/src/properties/url/url.tsx
--- a/webapp/src/properties/url/url.tsx	(revision cb21d48838b9db1750ffdfd44beee135d270e74f)
+++ b/webapp/src/properties/url/url.tsx	(date 1661622447568)
@@ -90,7 +90,7 @@
         <div className={`URLProperty ${props.property.valueClassName(props.readOnly)}`}>
             <a
                 className='link'
-                href={Utils.ensureProtocol((props.propertyValue as string).trim())}
+                href={isEmpty ? '' : Utils.ensureProtocol((props.propertyValue as string).trim())}
                 target='_blank'
                 rel='noreferrer'
                 onClick={(event) => event.stopPropagation()}
```

But in general it should be also fixed in the typescript types because `props.propertyValue` has type `string | string[]` which is actually not true now because we have `undefined` here also.

@jespino, @Pinjasaur what do you think about the fixing typescript types?

**Comment by Pinjasaur** [comment]
@kamre, agree the types should be updated. I see a couple options:

1. making `propertyValue` optional i.e. implicit `undefined`
2. explicitly defining an `undefined` value

Unless @jespino has strong opinions otherwise, I'm going to queue this up for discussion in our weekly dev meeting. That's public, you're welcome to join. https://community.mattermost.com/boards/team/rcgiyftm7jyrxnma1osd8zswby/886d2280-e88e-4065-8c85-1b62578f192e/f30dcb19-a0e8-45d2-8768-a9489f6e7fe8/cgayja6f4fffmur9urowe364qxw

**Comment by Rajat-Dabade** [comment]
@Pinjasaur @kamre, the default value of the `string` in typescript is `undefined` and for `string[]` it is an empty array i.e. `[]`, I guess the fix for this would be simple `if (propertyValue === undefined)` check at the place where all the properties are getting disturbed.

cc @jespino

**Comment by Pinjasaur** [comment]
@kamre you should be unblocked now. @jespino is planning on a more exhaustive update soon to better reflect how `propertyValue` works post-refactor.

**Comment by kamre** [comment]
@Pinjasaur @Rajat-Dabade thank you very much!
I was able to update my personal focalboard server to latest `main` after fixing migrations manually.
